### PR TITLE
Mocking Frameworks

### LINKED ISSUE #13: Mocking Frameworks
We use quite a few mocking frameworks

- [NSubstitute](http://nsubstitute.github.io/)
- [Moq](https://github.com/Moq/moq4/wiki/Quickstart)
- [Rhino Mocks](https://hibernatingrhinos.com/oss/rhino-mocks)

Where should they be on the Tech Radar?

### PR DESCRIPTION
**Pull request by fffej**
This could resolve #13 

Preview at https://radar.thoughtworks.com/?sheetId=https%3A%2F%2Fraw.githubusercontent.com%2Fred-gate%2FTech-Radar%2Fbdfc23e5ecee31a3351836f316dde657a6317927%2Fradar.csv

### CODE REVIEW

File: radar.csv [hunk 1]
```diff
@@ -24,3 +24,6 @@ Edge,adopt,platforms,true,"Edge is Microsoft's evergreen browser. We support it
 IE 11,adopt,platforms,true,"IE 11 is the default browser on Windows 10 and makes up approximately 10% of the visits to Redgate.com. We support it by default"
 IE 10 and under,retire,platforms,true,"We do not support IE10 and under"
 Raven DB,retire,tools,true,"We previously used Raven DB in DLM Dashboard but our experience was not positive."
+NSubsitute,adopt,languages and frameworks,true,"NSubstitute is our default mocking library of choice. We feel it's got less setup cost than Moq and requires fewer .Object calls"
+Moq,endure,languages and frameworks,true,"Moq is our legacy framework for Moq. We see no benefit in having a common mocking framework across solutions"
```
Comment (nyctef) [inline radar.csv]: Minor typo: "Moq is our legacy framework for Moq"

"Legacy" feels like a bit of a strong word here, but the sentiment is correct

File: radar.csv [hunk 2]
```diff
@@ -24,3 +24,6 @@ Edge,adopt,platforms,true,"Edge is Microsoft's evergreen browser. We support it
...
+Fakes,explore,techniques,true,"We've experimented using Fakes in the Redgate Usage Client - fakes used in the test can reduce the amount of mocking is required and can make the tests simpler"
```
Comment (nyctef) [inline radar.csv]: This section would probably benefit from a definition of what exactly we mean by fakes and mocking (personal preference is from here: https://github.com/testdouble/contributing-tests/wiki/Test-Double - in particular the two definitions for "Fake")

**Comment by garethbragg** [comment]
I did a quick rebase to get this merge-able:

https://radar.thoughtworks.com/?sheetId=https://raw.githubusercontent.com/red-gate/Tech-Radar/9becdac9f0bf1af1618a75089b001deb372b366d/radar.csv

**Comment by fffej** [comment]
Just a prod of @nyctef / @samblackburn to review and merge if all is good 😄

**Comment by nyctef** [comment]
We could potentially still make more of the distinction between "fakes" as "hand-rolled alternatives to a mocking library" ([which is what Figleaf has](https://github.com/red-gate/data-masking/blob/c5f0f5266a8c1e4498a01aea45d3cb71b12c9c7c/Figleaf/Backend/Figleaf.Core.Tests/Utils/FakeRowCountQuery.cs)) and "a full in-memory implementation of an interface" ([which is what the usage client has](https://github.com/red-gate/RedGate.Usage.Client/blob/dc3f3b60c64e36eca65cabc63f0e3b03c1fdc458/RedGate.Usage.Client.Tests/Providers/Registry/FakeRegistryProvider.cs)). For new projects I think you generally want to start off hand-rolling everything and then introduce a mocking library if/when that becomes painful, but you might always need proper fake implementations of certain interfaces.

That's just nitpicking at this point, though - I think this is fine to merge 👍

**Pull request merged**